GenumKit/Parsers/AssetsCatalogParser.swift

```diff
@@ -20,14 +21,67 @@ public final class AssetsCatalogParser {
     }
   }
 
-  public func parseDirectory(path: String) {
-    if let dirEnum = NSFileManager.defaultManager().enumeratorAtPath(path) {
-      while let path = dirEnum.nextObject() as? NSString {
-        if path.pathExtension == "imageset" {
-          let imageName = (path.lastPathComponent as NSString).stringByDeletingPathExtension
-          self.addImageName(imageName)
+  public func parseCatalog(path: String) {
+    guard let items = loadAssetCatalogContents(path) else { return }
+
+    // process recursively
+    process(items)
+  }
+}
+
+// MARK: - Plist processing
+
+private enum AssetCatalog: String {
+  case children = "children"
+  case filename = "filename"
+  case providesNamespace = "provides-namespace"
+  case root = "com.apple.actool.catalog-contents"
+}
+
+extension AssetsCatalogParser {
+  static let ImageSet = "imageset"
```

Comment:
I'd rename that constant `imageSetExtension` (*lower*CsmelCase seems to be the rule even for constants in Swift I think)